src/plugin.ts

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+import fs = require('fs');
+import path = require('path');
+import sizeTree = require('./size_tree');
+
+export class WebpackBundleSizeAnalyzerPlugin {
+  filepath: string;
+  statsOptions: object;
+  constructor(filepath: string = '', statsOptions: object = {}) {
+    this.filepath = filepath;
+    this.statsOptions = statsOptions || {};
+  }
+  apply(compiler: any) {
+    const that = this;
```

Comment:
You were spot on with your review here.  I think I was running into other issues when I tried the `that = this` hack (and forgot to remove it).